cmd/celestia-appd/cmd/start.go

```diff
@@ -0,0 +1,627 @@
+	dbm "github.com/tendermint/tm-db"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+)
+
+const (
+	// Tendermint full-node start flags
+	flagWithTendermint = "with-tendermint"
+	flagAddress        = "address"
+	flagTransport      = "transport"
+	flagTraceStore     = "trace-store"
+	flagCPUProfile     = "cpu-profile"
+
+	FlagForceNoBBR = "force-no-bbr"
+
+	// gRPC-related flags
+	flagGRPCOnly       = "grpc-only"
+	flagGRPCEnable     = "grpc.enable"
+	flagGRPCAddress    = "grpc.address"
+	flagGRPCWebEnable  = "grpc-web.enable"
+	flagGRPCWebAddress = "grpc-web.address"
+)
+
+// startCmd runs the service passed in, either stand-alone or in-process with
+// Tendermint.
+func startCmd(appCreator srvrtypes.AppCreator, defaultNodeHome string) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "start",
+		Short: "Run the full node",
+		Long: `Run the full node application with Tendermint in or out of process. By
+default, the application will run with Tendermint in process.
+
+Pruning options can be provided via the '--pruning' flag or alternatively with '--pruning-keep-recent', and
+'pruning-interval' together.
+
+For '--pruning' the options are as follows:
+
+default: the last 362880 states are kept, pruning at 10 block intervals
+nothing: all historic states will be saved, nothing will be deleted (i.e. archiving node)
+everything: 2 latest states will be kept; pruning at 10 block intervals.
+custom: allow pruning options to be manually specified through 'pruning-keep-recent', and 'pruning-interval'
+
+Node halting configurations exist in the form of two flags: '--halt-height' and '--halt-time'. During
+the ABCI Commit phase, the node will check if the current block height is greater than or equal to
+the halt-height or if the current block time is greater than or equal to the halt-time. If so, the
+node will attempt to gracefully shutdown and the block will not be committed. In addition, the node
+will not be able to commit subsequent blocks.
+
+For profiling and benchmarking purposes, CPU profiling can be enabled via the '--cpu-profile' flag
+which accepts a path for the resulting pprof file.
+
+The node may be started in a 'query only' mode where only the gRPC and JSON HTTP
+API services are enabled via the 'grpc-only' flag. In this mode, Tendermint is
+bypassed and can be used when legacy queries are needed after an on-chain upgrade
+is performed. Note, when enabled, gRPC will also be automatically enabled.
+`,
+		PreRunE: func(cmd *cobra.Command, _ []string) error {
+			serverCtx := server.GetServerContextFromCmd(cmd)
+
+			// Bind flags to the Context's Viper so the app construction can set
+			// options accordingly.
+			if err := serverCtx.Viper.BindPFlags(cmd.Flags()); err != nil {
+				return err
+			}
+
+			_, err := server.GetPruningOptionsFromFlags(serverCtx.Viper)
+			return err
+		},
+		RunE: func(cmd *cobra.Command, _ []string) error {
+			err := checkBBR(cmd)
+			if err != nil {
+				return err
+			}
```

Comment:
besides adding imports and not exporting a few functions that previously were, this is the only change to the code :upside_down_face: